### PR TITLE
MC-115 & MC-116 - SonicWall Firewall - Check Host Status - Slack and Microsoft Teams

### PR DESCRIPTION
**Pull request by dmandyna-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

#### Icon-Validate

<details>

```
[*] Validating workflow at ./Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 40.824ms
```

<summary>
icon-validate ./Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/
</summary>
</details>

<details>

```
[*] Validating workflow at ./Check_Host_Block_Status_with_SonicWall_Firewall_from_Microsoft_Teams/

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 40.268ms
```

<summary>
icon-validate ./Check_Host_Block_Status_with_SonicWall_Firewall_from_Microsoft_Teams/
</summary>
</details>

### Screenshots

<details>

##### Microsoft Teams
workflow1

workflow2

##### Slack
workflow1

workflow2

<summary>
Workflow Builder
</summary>
</details>
<details>

##### Microsoft Teams
teams_message1

##### Slack
slack_message1

<summary>
Messages
</summary>
</details>

<details>

##### Microsoft Teams
artifact1

##### Slack
artifact1

<summary>
Artifact
</summary>
</details>

### CODE REVIEW

File: workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/workflow.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+name: Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack
+title: "Check Host Block Status with SonicWall Firewall from Slack"
+description: This workflow accepts a Slack command and checks if an IP address or domain is blocked, and posts the results in a message thread.
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [remediation_management, threat_detection_and_response]
+  keywords: [sonicwal, sonicos, firewall, slack, chatops, work_from_home]
```
Comment (ncramer-r7) [inline workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/workflow.spec.yaml:12]:
```suggestion
  keywords: [sonicwall, sonicos, firewall, slack, chatops, work_from_home]
```

File: workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Microsoft_Teams/workflow.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+name: Check_Host_Block_Status_with_SonicWall_Firewall_from_Microsoft_Teams
+title: "Check Host Block Status with SonicWall Firewall from Microsoft Teams"
+description: This workflow accepts a Microsoft Teams command and checks if an IP address or domain is blocked, and posts the results in a message thread.
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [remediation_management, threat_detection_and_response]
+  keywords: [sonicwal, sonicos, firewall, teams, microsoft_teams, chatops, work_from_home]
```
Comment (ncramer-r7) [inline workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Microsoft_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [sonicwall, sonicos, firewall, teams, microsoft_teams, chatops, work_from_home]
```

**Review by ncramer-r7 — CHANGES_REQUESTED** [review]
Other than the very small changes in the workflow.spec.yaml files, looks good to me!

File: workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/help.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Sample Slack Trigger Commands:
+
+`@Rapid7 InsightConnect block-status 198.51.100.100 198.51.100.101`
+
+`@Rapid7 InsightConnect block-status example.com example2.com 198.50.100.1`
+
+# Key Features
+
+* The ability to see if an IP is currently blocked by the firewall
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_SonicWall_Firewall_from_Slack/help.md:13]:
```suggestion
* The ability to see if an IP address is currently blocked by the firewall
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Comment by dmandyna-r7** [comment]
Hey @mrinehart-r7 / @ncramer-r7!
I've applied all the suggestions, thank you for the review! 😃

**Review by mrinehart-r7 — APPROVED** [review]

**Review by ncramer-r7 — APPROVED** [review]

**Pull request merged**